docs/hpc/06_tools_and_software/02_conda_environments.md

```diff
@@ -1 +1,268 @@
+
+Thus you can consider the following options
+
+-   Reinstall your packages if some of the files get deleted
+    -   You can do this manually 
+    -   You can do this automatically. For example, within a workflow of a pipeline software like [Nextflow](https://www.nextflow.io/)
+-   Pay for "Research Project Space" - read more [here](../03_storage/05_research_project_space.md)
+-   Use Singularity and install packages within a corresponding overlay file - read more [here](../07_containers/03_singularity_with_conda.md)  
+
+## Python
+Load anaconda module
+```sh
+module purge
+module load anaconda3/2020.07
+```
+:::tip
+Keep your program/project in `/scratch` and create conda environment using '-p' parameter. This will keep all the files inside the project's directory, instead of putting in in your `/home/$USER`
+:::
+
+```sh
+conda create -p ./penv python=3  ## environment will be created in project directory
+conda activate ./penv 
+```
+Also, you need to create a symbolic link, so conda will download files for packages to be installed into scratch, not your home directory.
+```sh
+mkdir /home/<NetID>/.conda
+mkdir /scratch/<NetID>/conda_pkgs
+ln -s /scratch/<NetID>/conda_pkgs /home/<NetID>/.conda/pkgs
+```
+[Install pre-compiled packages available in conda](https://anaconda.org/anaconda/repo)
+```sh
+conda install -c anaconda pandas
+```
+
+Other packages may be installed (and compiled when needed) using pip
+```sh
+pip install <package_name>
+```
+:::note
+Conda and packages install by default to `~/.local/lib/python<version>`
+:::
+
+If you did use `'pip install --user'` to install some packages (without conda or other virtual environment), they will be available in `~/.local/lib/python<version>`
+
+:::warning
+***The primary takeaway:***
+
+Let say you have tornado v.6 installed in `~/.local/lib/python<version>`, and tornado v.5 installed by `conda install`.
+
+When you will do `conda activate` you will have tornado v.6 available!! Not v.5!!
+
+(this behaviour is the same for packages installed by to `~/.local/lib/python<version>` before or after you create your conda environment)
+
+`pip freeze` will give v.6
+
+`conda list` will give v.5
+
+***Solution***
+
+To overcome this, do `export PYTHONNOUSERSITE=True` after conda activate
+:::
+
+## R
+Load anaconda module
+```sh
+module load anaconda3/2020.07
+```
+:::tip
+Keep your program/project in `/scratch` and create conda environment using '-p' parameter. This will keep all the files inside the project's directory, instead of putting them in your `/home/$USER`
+:::
+
+```sh
+conda create -p ./renv r=3.5 ## environment will be created in project directory
```

Comment:
Maybe have the examples use current versions of R, i.e. r-4.4?

Reply:
Oh, yeah, that's a great point.  Thanks for pointing it out.  Will do.